### PR TITLE
fix(storage): fix wrong parameter of hit_sstable_bloom_filter

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

After merge https://github.com/risingwavelabs/risingwave/pull/6130, we found some wrong when using state_store_v1 backend. 

This PR fix wrong parameter of `hit_sstable_bloom_filter` and reduce `UserKey::encode` times

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- fix wrong parameter of `hit_sstable_bloom_filter`
- remove `UserKey::for_test` in state_store 
- reduce `UserKey::encode` times

## Checklist

- ~[ ] I have written necessary rustdoc comments~
- ~[ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- https://github.com/risingwavelabs/risingwave/issues/6353

### CODE REVIEW

**Review by Gun9niR — APPROVED** [review]
LGTM. Sorry for the trouble!!

File: src/storage/src/hummock/store/version.rs [hunk 1]
```diff
@@ -503,18 +503,23 @@ impl HummockVersionReader {
             staging_iters.push(HummockIteratorUnion::First(imm.into_forward_iter()));
         }
         let mut staging_sst_iter_count = 0;
+        // encode once
+        let prefix_hint = if let Some(prefix) = read_options.prefix_hint.as_ref() {
```
Comment (wenym1) [inline src/storage/src/hummock/store/version.rs:507]: Can we use Some(...) and None for prefix_hint?

File: src/storage/src/hummock/state_store_v1.rs [hunk 2]
```diff
@@ -298,6 +298,12 @@ impl HummockStorageV1 {
             user_key_range.1.as_ref().map(UserKey::encode),
         );
         assert!(pinned_version.is_valid());
+        // encode once
```
Comment (wenym1) [inline src/storage/src/hummock/state_store_v1.rs:301]: ditto

**Review by wenym1 — APPROVED** [review]
Rest LGTM

File: src/storage/src/hummock/store/version.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/storage/src/hummock/store/version.rs:507]: +1

**Pull request merged**